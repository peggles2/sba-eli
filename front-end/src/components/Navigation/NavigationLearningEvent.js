import React, { Component } from "react";
import axios from "axios";
import { Dropdown } from "semantic-ui-react";
import NavigationLearningEventItem from "./NavigationLearningEventItem";

export default class NavigationLearningEvent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      learningEvents: []
    };
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    const url = process.env.REACT_APP_SERVICE_HOST + `/learning_events/`;

    const eventParams = {
      course_id: this.props.learningPathId,
      module_id: this.props.learningObjectiveId
    };

    axios
      .get(url, { params: eventParams })
      .then(res => {
        const learningEvents = res.data;
        this.setState({ learningEvents });
      })
      .catch(error => {
        console.log(error);
      });
  }

  renderLearningEvents(learningEvents = []) {
    if (learningEvents.length) {
      const course_id = this.props.learningPathId;
      const module_id = this.props.learningObjectiveId;
      return learningEvents.map(event => {
        const path =
          `/learning_paths/${course_id}` +
          `/learning_objectives/${module_id}` +
          `/learning_events/${event.id}`;
        return (
          <NavigationLearningEventItem
            key={event.id + course_id + module_id}
            path={path}
            name={event.title}
          />
        );
      });
    }

    return <NavigationLearningEventItem path="/" name="No data" />;
  }

  render() {
    return (
      <Dropdown item text={this.props.name}>
        <Dropdown.Menu>
          {this.renderLearningEvents(this.state.learningEvents)}
        </Dropdown.Menu>
      </Dropdown>
    );
  }
}
