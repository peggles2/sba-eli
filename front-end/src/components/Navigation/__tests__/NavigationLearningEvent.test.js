import React from 'react';
import NavigationLearningEvent from '../NavigationLearningEvent'
import { shallow } from 'enzyme';

describe('NavigationLearningEvent', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavigationLearningEvent/>);
    
    expect(wrapper).toMatchSnapshot();
  });

});
