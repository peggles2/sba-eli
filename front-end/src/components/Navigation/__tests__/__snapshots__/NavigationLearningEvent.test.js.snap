// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLearningEvent should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationLearningEvent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "additionLabel": "Add ",
      "additionPosition": "top",
      "children": <DropdownMenu>
        <NavigationLearningEventItem
          name="No data"
          path="/"
        />
      </DropdownMenu>,
      "closeOnBlur": true,
      "deburr": false,
      "icon": "dropdown",
      "item": true,
      "minCharacters": 1,
      "noResultsMessage": "No results found.",
      "openOnFocus": true,
      "renderLabel": [Function],
      "searchInput": "text",
      "selectOnBlur": true,
      "selectOnNavigation": true,
      "text": undefined,
      "wrapSelection": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <NavigationLearningEventItem
          name="No data"
          path="/"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "No data",
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "additionLabel": "Add ",
        "additionPosition": "top",
        "children": <DropdownMenu>
          <NavigationLearningEventItem
            name="No data"
            path="/"
          />
        </DropdownMenu>,
        "closeOnBlur": true,
        "deburr": false,
        "icon": "dropdown",
        "item": true,
        "minCharacters": 1,
        "noResultsMessage": "No results found.",
        "openOnFocus": true,
        "renderLabel": [Function],
        "searchInput": "text",
        "selectOnBlur": true,
        "selectOnNavigation": true,
        "text": undefined,
        "wrapSelection": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <NavigationLearningEventItem
            name="No data"
            path="/"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "No data",
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
